packages/strapi-plugin-i18n/config/functions/migrations/field.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+const { difference, pick, orderBy, prop, intersection } = require('lodash/fp');
+const { getService } = require('../../../utils');
+
+const getSubstitute = arr => arr.map(() => '??').join(', ');
+
+const before = () => {};
+
+// Migration when i18n is disabled on a field of a content-type that have i18n enabled
+const after = async ({ model, definition, previousDefinition, ORM }) => {
+  const ctService = getService('content-types');
+  const localeService = getService('locales');
+
+  if (!ctService.isLocalized(model)) {
+    return;
+  }
+
+  const localizedAttributes = ctService.getLocalizedFields(definition);
+  const prevLocalizedAttributes = ctService.getLocalizedFields(previousDefinition);
+  const attributesDisabled = difference(prevLocalizedAttributes, localizedAttributes);
+  const attributesToMigrate = intersection(Object.keys(definition.attributes), attributesDisabled);
+
+  if (attributesToMigrate.length === 0) {
+    return;
+  }
+
+  let locales = await localeService.find();
+  locales = await localeService.setIsDefault(locales);
+  locales = orderBy(['isDefault', 'code'], ['desc', 'asc'])(locales); // Put default locale first
+
+  const processedLocaleCodes = [];
+
+  if (model.orm === 'bookshelf') {
+    const trx = await ORM.knex.transaction();
+    try {
+      const columnsToCopy = ['id', ...attributesToMigrate];
+
+      await trx.raw('DROP TABLE IF EXISTS __tmp__i18n_field_migration');
+      await trx.raw(
+        `CREATE TABLE __tmp__i18n_field_migration AS SELECT ${getSubstitute(
```

Comment:
I would recommend using knex.schema to make sure we are compatible with all knex supported tables.

We do something similar for the sqlite rebuild:

```js
// create empty table
knex.schema.createTable()

// insert
knex.insert(qb => qb.select(...columns).from(model.collectionName)).into(tmpTable)

```

Reply:
Actually what we do with sqlite is not just `knex.schema.createTable`, we compute and create the table based on the model attributes (to use correct column types). To do that here too, I would need to pass the function `createTable` to the migration function. Is it what you have in mind?

Reply:
Not super super straight forward indeed...
You have the definition of the model so you can do it but need to use one of the util from the connectors to transform field type into column type. 


something simpler we can do is 

```js
const q = await knex.raw('CREATE TABLE __tmp__ AS ??', [
    knex.select(...cols).from('test_table'),
]);
```
I'm just not super confiendt with the raw part being agnostic. We must add tests that will run for each DB for that